### PR TITLE
[Core] Remove the unused MetricPointExporter

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
`MetricPointExporter` internally uses `StdoutExporterClient` which is a no-op so it doesn't export anything. We are effectively only using `OpenCensusProtoExporter` to export metrics to metrics agent.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/stats/metric_exporter_client.cc [hunk 1]
```diff
@@ -1,51 +0,0 @@
...
-
-#include <algorithm>
-
-namespace ray {
-namespace stats {
-
-///
-/// Stdout Exporter
-///
-void StdoutExporterClient::ReportMetrics(const std::vector<MetricPoint> &points) {}
```
Comment (jjyao) [inline src/ray/stats/metric_exporter_client.cc:25]: This is no-op.

File: src/ray/stats/stats.h [hunk 2]
```diff
@@ -104,7 +90,6 @@ static inline void Init(
   opencensus::stats::DeltaProducer::Get()->SetHarvestInterval(
       StatsConfig::instance().GetHarvestInterval());
 
-  MetricPointExporter::Register(exporter, metrics_report_batch_size);
```
Comment (rickyyx) [inline src/ray/stats/stats.h:107]: IIRC i remember u had a PR that changes this?  @rkooo567 
So this was not used?

**Review by rickyyx — COMMENTED** [review]
Just one question to @rkooo567 

<div><br/>via <a href="https://giphy.com/halloweenparty/">Halloween Party</a> on <a href="https://giphy.com/gifs/halloweenparty-halloween-party-costume-xVudSZL97djWljDzvg">GIPHY</a></div>

**Comment by jjyao** [comment]
@larrylian @wumuzi520 @SongGuyang is this something still used by Ant?

**Comment by larrylian** [comment]
@jjyao  Yes, this is still used by Ant.  This is very important. Ant Group uses a time-series database called CeresDB. By extending MetricExporterDecorator, we have implemented own OpenCensusProtoExporter that reports metric data via the HTTP protocol.
```
auto ceresdb_exporter = InitCeresdbExporter(exporter);
MetricPointExporter::Register(ceresdb_exporter, metrics_report_batch_size);
```

If you want to remove it and use a unified Exporter for reporting metric data, you need to refactor to allow other developers to customize the exporter client.

**Review by larrylian — CHANGES_REQUESTED** [review]
This is used by other developer

**Comment by jjyao** [comment]
@larrylian Got it.

I think the customization interface should be `opencensus::stats::StatsExporter::Handler` instead of `MetricExporterClient`. Like if someone wants to implement a different exporter, they should implement a subclass of `opencensus::stats::StatsExporter::Handler` instead of `MetricExporterClient`. `OpenCensusProtoExporter` is the default exporter implemented in Ray that exports metrics to the metrics agent. `MetricPointExporter` should be a customized one developed by Ant (the code should be in Ant's repo).

`ray::stats::Init` interface can be 

```
ray::stats::Init(
....
std::shared_ptr<opencensus::stats::StatsExporter::Handler> exporter_to_use = nullptr,
) {
  if exporter_to_use == nullptr:
     // Use the default exporter provided by Ray
     exporter_to_use = std::make_shared<>(new OpenCensusProtoExporter())
}
```


In the Ant's code, you can do `ray::stats::Init(exporter_to_use=MetricPointCeresDBExporter())`


cc @rkooo567 @rickyyx

**Comment by larrylian** [comment]
@jjyao  But if MetricPointExporter is deleted , `exporter_to_use`  will  not be used by `OpenCensusProtoExporter`.  
image

**Comment by jjyao** [comment]
> But if MetricPointExporter is deleted , exporter_to_use will not be used by OpenCensusProtoExporter

@larrylian `OpenCensusProtoExporter` doesn't use `exporter_to_use` (see your screenshot) even before this PR. `OpenCensusProtoExporter` directly implements the `opencensus::stats::StatsExporter::Handler` interface and has nothing to do with `MetricPointExporter`

**Comment by larrylian** [comment]
@jjyao 
Your current solution, I feel, would require other developers to write a lot more code. The original solution was to have `MetricPointExporter` extend `opencensus::stats::StatsExporter::Handler`, and then parse the `opencensus::stats::ViewData` data format into the `MetricPoint` format. Subsequently, other developers only need to implement the `MetricExporterClient`, responsible for reporting `MetricPoint` data to other time-series databases. If the `MetricPointExporter `is now removed, then developers will need to re-implement a similar functionality to `MetricPointExporter`.

Additionally, I want to mention the implementation of `OpenCensusProtoExporter`. I think it could totally inherit from `MetricExporterClient`, converting `MetricPoint` into `pc::ReportOCMetricsRequest` and then reporting it to the time-series database.

**Comment by jjyao** [comment]
@larrylian Is `MetricPoint` a standard format? One issue I found with `MetricPoint` is that it only contains subset of the data of `opencensus::stats::ViewData` (i.e. when you convert `opencensus::stats::ViewData` to `MetricPoint`, some data is lost, especially for the `Distribution` data) and that's why `OpenCensusProtoExporter` doesn't inherit from `MetricExporterClient`. Given `opencensus::stats::ViewData` is the source-of-truth, I think every exporter should directly consume, process and export it.

Happy to chat offline on this.

**Comment by larrylian** [comment]
@jjyao  OK， Let's discuss this offline.

**Comment by rickyyx** [comment]
LG on my end - leaving it to @larrylian if we wanna do this.

**Review by larrylian — APPROVED** [review]

**Comment by larrylian** [comment]
LGTM.
We have reached a consensus offline and now we will unify the approach using the extension of the `opencensus::stats::StatsExporter::Handler` interface.

**Review by rickyyx — APPROVED** [review]

**Pull request merged**